File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task105_GEMINI_gemini-1.5-pro-001.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <openssl/ssl.h>
#include <openssl/err.h>

int main(int argc, char **argv) {
    if (argc != 3) {
        fprintf(stderr, "Usage: %s <hostname> <port>
", argv[0]);
        return 1;
    }

    const char *hostname = argv[1];
    int port = atoi(argv[2]);

    SSL_CTX *ctx;
    SSL *ssl;
    BIO *bio;

    SSL_library_init();
    SSL_load_error_strings();
    ctx = SSL_CTX_new(TLS_client_method());
    if (!ctx) {
        fprintf(stderr, "Error creating SSL context
");
        return 1;
    }

    ssl = SSL_new(ctx);
    if (!ssl) {
        fprintf(stderr, "Error creating SSL object
");
        SSL_CTX_free(ctx);
        return 1;
    }

    bio = BIO_new_connect(hostname);
    if (!bio) {
        fprintf(stderr, "Error creating BIO object
");
        SSL_free(ssl);
        SSL_CTX_free(ctx);
        return 1;
    }

    BIO_set_conn_port(bio, port);
    SSL_set_bio(ssl, bio, bio);

    if (SSL_connect(ssl) <= 0) {
        fprintf(stderr, "Error establishing SSL connection
");
        ERR_print_errors_fp(stderr);
        SSL_free(ssl);
        SSL_CTX_free(ctx);
        BIO_free_all(bio);
        return 1;
    }

    printf("Secure connection established:
");
    printf("SSL Session ID: %s
", SSL_SESSION_get_id(SSL_get_session(ssl)));

    SSL_shutdown(ssl);
    SSL_free(ssl);
    SSL_CTX_free(ctx);
    BIO_free_all(bio);

    return 0;
}

File: dataset/output-bk/GEMINI_gemini-1.5-pro-001/Task105_GEMINI_gemini-1.5-pro-001.java
import javax.net.ssl.SSLSocket;
import javax.net.ssl.SSLSocketFactory;
import java.io.IOException;

public class SecureConnection {

    public static void main(String[] args) {
        if (args.length != 2) {
            System.err.println("Usage: java SecureConnection <hostname> <port>");
            return;
        }

        String hostname = args[0];
        int port = Integer.parseInt(args[1]);

        try {
            SSLSocketFactory factory = (SSLSocketFactory) SSLSocketFactory.getDefault();
            SSLSocket socket = (SSLSocket) factory.createSocket(hostname, port);

            System.out.println("Secure connection established:");
            System.out.println("Socket: " + socket);
            System.out.println("Session: " + socket.getSession());

            socket.close();

        } catch (IOException e) {
            System.err.println("Error creating secure connection: " + e.getMessage());
        }
    }
}